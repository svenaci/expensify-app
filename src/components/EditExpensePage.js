import React from "react";
const EditExpensePage = () => <div>Ths is my edit expense componen of</div>;

export default EditExpensePage;
