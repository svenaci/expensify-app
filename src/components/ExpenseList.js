import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem.js";

const ExpenseList = (props) => (
  <div>
    <h1>Expense list</h1>
    {props.expenses.map((expense) => {
      return <ExpenseListItem key={expense.id} {...expense} />;
    })}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
  };
};

export default connect(mapStateToProps)(ExpenseList);
