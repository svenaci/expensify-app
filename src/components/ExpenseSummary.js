import React from "react";
import { connect } from "react-redux";
import moment from "moment";
import numeral from "numeral";
import getExpensesTotal from "../selectors/expensesTotal";
import getVisibleExpenses from "../selectors/expenses";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal / 100).format("$0,0.00");

  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} to {formattedExpensesTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
