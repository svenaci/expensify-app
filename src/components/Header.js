import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <NavLink className="header__title" to="/" exact={true}>
          <h1>Expense App</h1>
        </NavLink>
        <NavLink to="/create" activeClassName="is-active">
          Create Expense
        </NavLink>
        <NavLink to="/help" activeClassName="is-active">
          Help
        </NavLink>
        <button>Logout</button>
      </div>
    </div>
  </header>
);

export default Header;
