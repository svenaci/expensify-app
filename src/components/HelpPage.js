import React from "react";

const HelpPage = () => <div>Ths is my help component</div>;

export default HelpPage;
