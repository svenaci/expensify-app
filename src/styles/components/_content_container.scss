.content-container {
  max-width: 80%; //80rem
  margin: 0 auto;
  padding: 0 $m-size;
}
