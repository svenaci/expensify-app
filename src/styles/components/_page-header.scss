.page-header {
  background: #f7f7f7;
  padding: $l-size 0;
  margin-bottom: $l-size;
}

.page-header__title {
  margin: 0;
  font-weight: 300;
}

.page-header__font {
  font-weight: bold;
}

.page-header__actions {
  margin-top: $m-size;
}
