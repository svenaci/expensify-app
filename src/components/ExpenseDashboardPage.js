import React from "react";
import ExpenseList from "./ExpenseList.js";
import ExpenseListFilters from "./ExpenseListFilters.js";
import ExpensesSummary from "./ExpenseSummary.js";

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
    <ExpensesSummary />
  </div>
);

export default ExpenseDashboardPage;
