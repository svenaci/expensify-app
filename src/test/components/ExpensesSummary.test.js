import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpenseSummary";

test("should render ExpensesSummary correctly", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={235} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary correctly", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expensesTotal={2350} />
  );
  expect(wrapper).toMatchSnapshot();
});
