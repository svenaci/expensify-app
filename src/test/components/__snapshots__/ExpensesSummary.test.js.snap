// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span
        className="page-header__font"
      >
        1
      </span>
       
      expense
       totalling to
       
      <span
        className="page-header__font"
      >
        $2.35
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary correctly 2`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span
        className="page-header__font"
      >
        2
      </span>
       
      expenses
       totalling to
       
      <span
        className="page-header__font"
      >
        $23.50
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
